toolkit/tools/imagecustomizerapi/config.go

```diff
@@ -64,5 +64,42 @@ func (c *Config) IsValid() (err error) {
 		return fmt.Errorf("os.resetBootLoaderType must be specified if resetPartitionsUuidsType is specified")
 	}
 
+	if c.OS != nil && c.OS.Verity != nil {
+		err := ensureVerityPartitionIdExists(c.OS.Verity.DataPartition, c.Storage)
+		if err != nil {
+			return fmt.Errorf("invalid verity 'dataPartition':\n%w", err)
+		}
+
+		err = ensureVerityPartitionIdExists(c.OS.Verity.HashPartition, c.Storage)
+		if err != nil {
+			return fmt.Errorf("invalid verity 'hashPartition':\n%w", err)
+		}
+	}
+
+	return nil
+}
+
+func ensureVerityPartitionIdExists(verityPartition IdentifiedPartition, storage *Storage) error {
+	switch verityPartition.IdType {
+	case IdTypeId:
+		if storage == nil {
+			return fmt.Errorf("'idType' cannot be 'id' if 'storage' is not specified")
+		}
```

Comment:
This can be moved outside and above the switch case so we exit immediately if `storage` is not provided.

Reply:
`storage` is only needed for `IdTypeId`. The other types can work even if `storage` is not provided.

Reply:
Understood, I will resolve this comment then.